From a0d1d068b921bb5566bbbf2b5fdbaf3a2fa823dd Mon Sep 17 00:00:00 2001
From: Balamurugan Selvarajan <quic_bselvara@quicinc.com>
Date: Wed, 18 May 2022 22:43:41 +0530
Subject: [PATCH] ath12k: Destroy debugfs during firmware recovery

During firmware recovery of core, destroy existing debugfs
files. This avoids below print
debugfs: Directory 'twt' with parent 'netdev:wlan0' already present!

Signed-off-by: Balamurugan Selvarajan <quic_bselvara@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/core.c |  7 +++++++
 drivers/net/wireless/ath/ath12k/mac.c  | 10 ++++++++++
 2 files changed, 17 insertions(+)

Index: backports-20220404-5.4.164-f40abb4788/drivers/net/wireless/ath/ath12k/core.c
===================================================================
--- backports-20220404-5.4.164-f40abb4788.orig/drivers/net/wireless/ath/ath12k/core.c
+++ backports-20220404-5.4.164-f40abb4788/drivers/net/wireless/ath/ath12k/core.c
@@ -812,6 +812,7 @@ static void ath12k_core_restart(struct w
 	int ret;
 	struct ath12k *ar;
 	struct ath12k_pdev *pdev;
+	struct ath12k_vif *arvif;
 	int i;
 
 	spin_lock_bh(&ab->base_lock);
@@ -824,6 +825,12 @@ static void ath12k_core_restart(struct w
 		if (!ar || ar->state == ATH12K_STATE_OFF)
 			continue;
 
+		list_for_each_entry(arvif, &ar->arvifs, list) {
+			if (arvif->is_started) {
+				ath12k_debugfs_remove_interface(arvif);
+			}
+		}
+
 		ieee80211_stop_queues(ar->hw);
 		ath12k_mac_drain_tx(ar);
 		complete(&ar->scan.started);
